armi/physics/fuelCycle/fuelHandlers.py

```diff
@@ -1174,18 +1175,24 @@ def buildRingSchedule(
         findAssembly
 
         """
+        maxRingInCore = self.r.core.getNumRings()
+        if dischargeRing > maxRingInCore:
+            runLog.warning(f"Discharge ring {dischargeRing} is outside the core (max {maxRingInCore}).")
```

Comment:
Can you include some further description in the warning for how the use may fix this? Same for the warning below.

Reply:
I tried. this method can be called by a variety of paths so I can 't be too specific.